pkg/streamingpromql/benchmarks/comparison_test.go

```diff
@@ -189,12 +189,9 @@ func TestBenchmarkSetup(t *testing.T) {
 
 // Why do we do this rather than require.Equal(t, expected, actual)?
 // It's possible that floating point values are slightly different due to imprecision, but require.Equal doesn't allow us to set an allowable difference.
-func requireEqualResults(t testing.TB, expected, actual *promql.Result) {
+func requireEqualResults(t testing.TB, expr string, expected, actual *promql.Result) {
 	require.Equal(t, expected.Err, actual.Err)
-
-	// Ignore warnings until they're supported by the streaming engine.
-	// require.Equal(t, expected.Warnings, actual.Warnings)
-
+	require.ElementsMatch(t, expected.Warnings.AsStrings(expr, 0), actual.Warnings.AsStrings(expr, 0))
```

Comment:
Why not check the position of the warnings as well? (ditto in some of the other tests)